src/types.tsx

```diff
@@ -26,3 +26,59 @@ export interface ThemeInterface {
 export interface InputFieldProps {
   hasError?: boolean;
 }
+
+export interface AddNewClassFormProps {
+  isOpen: boolean;
+  toggle(): void;
+  fetchClasses(): void;
+}
+
+export interface AddNewClassWorkFormProps {
+  classId: string | undefined;
+  isOpen: boolean;
+  toggle(): void;
+  fetchClasses(): void;
+}
+
+export interface deleteClassProps {
+  className: string;
+  classID: string;
+  isOpen: boolean;
+  toggle(): void;
+  fetchClasses(): void;
+}
+
+export interface ClassComponentDataProps {
+  _id: string;
+  createdAt: string;
+  name: string;
+  description: string;
+  classworks?: [];
+}
+export interface EditClassModalProps {
+  item: ClassComponentDataProps;
+  isOpen: boolean;
+  toggle(): void;
+  fetchClasses(): void;
+  class?: string | undefined;
+}
+
+export interface classesProps {
+  classes: ClassComponentDataProps[] | undefined;
+  fetchClasses(): void;
+}
+
+export interface headerProps {
+  item: ClassComponentDataProps;
+  fetchClasses(): void;
+}
+
+export interface classesWorkProps {
+  name: string;
+  description: string;
+}
+
+export interface ClassHeaderDataProps {
+  item: ClassComponentDataProps;
+  fetchClasses(): void;
+}
```

Comment:
I don't think these should be here since they're very specific to the class component props. We should only put the interface/types in this file that are used by multiple components

Reply:
moved in class folder